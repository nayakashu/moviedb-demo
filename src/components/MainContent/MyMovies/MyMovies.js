import React, { Component } from 'react';
import { connect } from 'react-redux';
import MovieTile from '../MovieTile/MovieTile';
import Loader from '../../common/Loader/Loader';
import { closeLeftNav, setActivePage } from '../../../actions/navActions';
import { fetchMyMovies } from '../../../actions/movieListActions';
import './MyMovies.scss';

export class MyMovies extends Component {
  componentDidMount() {
    this.props.fetchMyMovies();
    this.props.closeLeftNav();
    this.props.setActivePage();
  }

  render() {
    return (
      <div className="top-rated-movies">
        <h3>My Movies</h3>
        {this.props.userMovies ? null : <Loader />}
        <div className="row">
          {this.props.userMovies &&
            this.props.userMovies.map((item, index) => (
              <div className="col-sm-4 col-xs-6" key={index}>
                <MovieTile movieDetails={item && item.data} editable={true} />
              </div>
            ))}
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  userMovies: state.movieListState.userMovies
});

export default connect(
  mapStateToProps,
  { closeLeftNav, fetchMyMovies, setActivePage }
)(MyMovies);
