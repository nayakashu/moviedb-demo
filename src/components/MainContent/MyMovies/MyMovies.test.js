import React from 'react';
import { shallow } from 'enzyme';
import { MyMovies } from './MyMovies';

describe('MyMovies', () => {
  const props = {
    fetchMyMovies: () => {},
    closeLeftNav: () => {},
    setActivePage: () => {}
  };
  const myMovies = shallow(<MyMovies {...props} />);

  it('renders correctly', () => {
    expect(myMovies).toMatchSnapshot();
  });
});
