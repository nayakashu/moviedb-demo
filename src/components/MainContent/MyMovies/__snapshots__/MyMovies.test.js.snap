// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyMovies renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MyMovies
    closeLeftNav={[Function]}
    fetchMyMovies={[Function]}
    setActivePage={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h3>
          My Movies
        </h3>,
        <Loader />,
        <div
          className="row"
        />,
      ],
      "className": "top-rated-movies",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "My Movies",
        },
        "ref": null,
        "rendered": "My Movies",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "row",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h3>
            My Movies
          </h3>,
          <Loader />,
          <div
            className="row"
          />,
        ],
        "className": "top-rated-movies",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "My Movies",
          },
          "ref": null,
          "rendered": "My Movies",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "row",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
